src/sdk_number.rs

```diff
@@ -0,0 +1,121 @@
+use std::fmt;
+use serde::{Serialize, Serializer};
+
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub enum SDKNumber {
+    Api26,
+    Api25,
+    Api24,
+    Api23,
+    Api22,
+    Api21,
+    Api20,
+    Api19,
+    Api18,
+    Api17,
+    Api16,
+    Api15,
+    Api14,
+    Api13,
+    Api12,
+    Api11,
+    Api10,
+    Api9,
+    Api8,
+    Api7,
+    Api6,
+    Api5,
+    Api4,
+    Api3,
+    Api2,
+    Api1,
+    ApiUnknown,
+}
+
+
+use self::SDKNumber::*;
+
+
+/// As per: https://developer.android.com/reference/android/os/Build.VERSION_CODES.html
+impl SDKNumber {
+    pub fn as_str(&self) -> &str {
+        match *self {
+            Api26 => "Oreo (26)",
+            Api25 => "Nougat MR1 (25)",
+            Api24 => "Nougat (24)",
+            Api23 => "Marshmallow (23)",
+            Api22 => "Lollipop MR1 (22)",
+            Api21 => "Lollipop (21)",
+            Api20 => "KitKat Watch (20)",
+            Api19 => "KitKat (19)",
+            Api18 => "Jelly Bean MR2 (18)",
+            Api17 => "Jelly Bean MR1 (17)",
+            Api16 => "Jelly Bean (16)",
+            Api15 => "Ice Cream Sandwich MR1 (15)",
+            Api14 => "Ice Cream Sandwich (14)",
+            Api13 => "Honeycomb MR2 (13)",
+            Api12 => "Honeycomb MR1 (12)",
+            Api11 => "Honeycomb (11)",
+            Api10 => "Gingerbread MR1 (10)",
+            Api9  => "Gingerbread (9)",
+            Api8  => "Froyo (8)",
+            Api7  => "Eclair MR1 (7)",
+            Api6  => "Eclair 0.1 (6)",
+            Api5  => "Eclair (5)",
+            Api4  => "Donut (4)",
+            Api3  => "Cupcake (3)",
+            Api2  => "Base1.1 (2)",
+            Api1  => "Base (1)",
+            ApiUnknown => "Unknown (?)",
+        }
+    }
+}
+
+impl From<u32> for SDKNumber {
+    fn from(version: u32) -> SDKNumber {
+        match version {
+            1  => Api1,
```

Comment:
It looks rather tedious to keep repeating this, but I wouldn't know how to create a macro for this.